agent/eventhandler/task_handler_types.go

```diff
@@ -28,9 +29,13 @@ type sendableEvent struct {
 
 	taskSent   bool
 	taskChange api.TaskStateChange
+
+	mu sync.RWMutex
```

Comment:
It seems like this is protecting `isContainerEvent`. If so, can you rename this to reflect that? Example: `isContainerEventLock`